server/mailet/mailetcontainer-camel/src/test/java/org/apache/james/mailetcontainer/impl/JamesMailetContextTest.java

```diff
@@ -214,6 +216,45 @@ public void sendMailShouldEnqueueEmailWithOtherStateWhenSpecified() throws Excep
         assertThat(mailArgumentCaptor.getValue().getState()).isEqualTo(other);
     }
 
+    @Test
+    public void sendMailShouldEnqueueEmailWithRootStateAndDelayWhenSpecified() throws Exception {
+        MailImpl mail = new MailImpl();
+        mail.setSender(mailAddress);
+        mail.setRecipients(ImmutableList.of(mailAddress));
+        mail.setMessage(MimeMessageBuilder.defaultMimeMessage());
+        testee.sendMail(mail, 5, TimeUnit.MINUTES);
+
+        ArgumentCaptor<Mail> mailArgumentCaptor = ArgumentCaptor.forClass(Mail.class);
+        ArgumentCaptor<Long> delayArgumentCaptor = ArgumentCaptor.forClass(Long.class);
+        ArgumentCaptor<TimeUnit> timeUnitArgumentCaptor = ArgumentCaptor.forClass(TimeUnit.class);
+        verify(spoolMailQueue).enQueue(mailArgumentCaptor.capture(), delayArgumentCaptor.capture(), timeUnitArgumentCaptor.capture());
+        verifyNoMoreInteractions(spoolMailQueue);
+
+        assertThat(mailArgumentCaptor.getValue().getState()).isEqualTo(Mail.DEFAULT);
```

Comment:
If you need several assertions, you should use `SoftAssertions`

1. Create the relevant junit class:

```
    @Rule
    public JUnitSoftAssertions softly = new JUnitSoftAssertions();
```

2. And then in the test with several assertThat: (those with a simple assertThat don't need it)

```
softly.assertThat(...)
softly.assertThat(...)
softly.assertThat(...)
....
```

It allows the display of all test error in a single run and makes debugging easier.